### PR TITLE
Upgrade to rails 5

### PR DESCRIPTION
**Pull request by rubenarakelyan**
This PR upgrades `email-alert-api` to Rails 5. See individual commits for more details.

Trello: https://trello.com/c/bPmOMCxn/80-upgrade-email-alert-api-to-a-supported-version-of-rails

### CODE REVIEW

File: app/controllers/notifications_controller.rb [hunk 1]
```diff
@@ -18,11 +18,11 @@ def show
 private
 
   def notification_params
-    params.slice(:subject, :from_address_id, :urgent, :header, :footer, :document_type,
+    params.permit!.to_h.slice(:subject, :from_address_id, :urgent, :header, :footer, :document_type,
```
Comment (gpeng) [inline app/controllers/notifications_controller.rb]: Could we assign `params.permit!.to_h` to variable to DRY this up a bit?

File: app/controllers/subscriber_lists_controller.rb [hunk 2]
```diff
@@ -42,11 +42,11 @@ def subscriber_list_params
 
   def find_exact_query_params
     {
-      tags: params.fetch(:tags, {}),
-      links: params.fetch(:links, {}),
-      document_type: params.fetch(:document_type, ""),
-      email_document_supertype: params.fetch(:email_document_supertype, ""),
-      government_document_supertype: params.fetch(:government_document_supertype, ""),
+      tags: params.permit!.to_h.fetch(:tags, {}),
```
Comment (gpeng) [inline app/controllers/subscriber_lists_controller.rb]: And the same DRY-ification here?

File: config/application.rb [hunk 3]
```diff
@@ -1,38 +1,25 @@
...
 
-# Pick the frameworks you want:
-require "active_model/railtie"
-require "active_record/railtie"
-require "action_controller/railtie"
-require "action_mailer/railtie"
-require "action_view/railtie"
-# require "sprockets/railtie"
-# require "rails/test_unit/railtie"
+require 'rails/all'
```
Comment (gpeng) [inline config/application.rb]: Should we exclude the unused bits as per the Rails 4 version?

**Comment by davidbasalla** [comment]
Looks good to me! Don't know enough about the Jenkinsfile to judge the change, but 👍  from me!

File: app/controllers/notifications_controller.rb (hunk 1 above)
Comment (rubenarakelyan) [inline app/controllers/notifications_controller.rb]: Yup

File: app/controllers/subscriber_lists_controller.rb (hunk 2 above)
Comment (rubenarakelyan) [inline app/controllers/subscriber_lists_controller.rb]: Also yup

File: config/application.rb (hunk 3 above)
Comment (rubenarakelyan) [inline config/application.rb]: Indeed, we can exclude `action_job` and `action_cable`.

**Pull request merged**